2022-12-19 16:26:51,263 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-19 16:27:41,493 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_cli_commands_example' with args {} on 1 hosts
2022-12-19 16:31:03,278 -  nornir.core -     INFO -        run() - Running task 'napalm_get_example' with args {} on 1 hosts
2022-12-19 16:31:53,019 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-19 16:32:27,382 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-19 16:54:54,788 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-19 16:55:18,900 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-19 17:56:14,326 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-19 17:58:50,624 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_cli_commands_example' with args {} on 1 hosts
2022-12-19 18:00:24,617 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-21 01:40:53,048 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-23 01:24:47,289 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_cli_commands_example' with args {} on 1 hosts
2022-12-24 23:44:48,068 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-24 23:46:01,816 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-24 23:46:53,432 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-24 23:47:36,664 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-24 23:48:27,523 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-24 23:49:49,201 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-24 23:49:50,083 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'nornir_napalm_get_example' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/devnet/pyhton_nornir/2023/5.nornir_napalm/5.3.nornir_napalm_get_result_parsing.py", line 18, in nornir_napalm_get_example
    rprint(config.result)    #6
NameError: name 'config' is not defined

2022-12-24 23:50:41,512 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-24 23:50:42,148 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'nornir_napalm_get_example' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/devnet/pyhton_nornir/2023/5.nornir_napalm/5.3.nornir_napalm_get_result_parsing.py", line 19, in nornir_napalm_get_example
    rprint(config.result["get_config"]) #7
NameError: name 'config' is not defined

2022-12-24 23:51:01,493 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_get_example' with args {} on 1 hosts
2022-12-24 23:51:02,381 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'nornir_napalm_get_example' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/devnet/pyhton_nornir/2023/5.nornir_napalm/5.3.nornir_napalm_get_result_parsing.py", line 20, in nornir_napalm_get_example
    rprint(config.result["get_config"]["running"])    #8
NameError: name 'config' is not defined

2022-12-24 23:52:13,846 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-24 23:52:34,710 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-25 03:26:45,740 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-25 03:30:22,717 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:30:52,170 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:31:42,991 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:32:48,212 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:33:43,082 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:35:07,786 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:36:06,379 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:37:01,159 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:44:05,664 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:44:35,461 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:45:11,598 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:46:02,436 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:46:55,958 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:47:25,303 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:48:01,747 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-26 01:48:28,256 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_structured_output_example' with args {} on 1 hosts
2022-12-27 13:44:29,179 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 13:44:31,225 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'napalm_configure' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/.local/lib/python3.10/site-packages/nornir_napalm/plugins/tasks/napalm_configure.py", line 35, in napalm_configure
    device.load_merge_candidate(filename=filename, config=configuration)
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 317, in load_merge_candidate
    return_status, msg = self._load_candidate_wrapper(
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 284, in _load_candidate_wrapper
    (return_status, msg) = self._scp_file(
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 733, in _scp_file
    return self._xfer_file(
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 802, in _xfer_file
    raise CommandErrorException(msg)
napalm.base.exceptions.CommandErrorException: SCP file transfers are not enabled. Configure 'ip scp server enable' on the device.

2022-12-27 13:44:31,226 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'nornir_napalm_send_config_exmaple' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/devnet/pyhton_nornir/2023/5.nornir_napalm/5.4.nornir_napalm_send_config.py", line 9, in nornir_napalm_send_config_exmaple
    task.run(task=napalm_configure, filename="config_file.txt")
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 174, in run
    raise NornirSubTaskError(task=run_task, result=r)
nornir.core.exceptions.NornirSubTaskError: Subtask: napalm_configure (failed)


2022-12-27 13:45:39,014 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 13:50:05,940 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_write_file_example' with args {} on 1 hosts
2022-12-27 13:50:51,239 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_write_file_example' with args {} on 1 hosts
2022-12-27 13:51:37,387 -  nornir.core -     INFO -        run() - Running task 'nornir_utils_write_file_example' with args {} on 1 hosts
2022-12-27 13:51:45,318 -  nornir.core -     INFO -        run() - Running task 'nornir_utils_write_file_example' with args {} on 1 hosts
2022-12-27 13:53:39,795 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 14:02:44,271 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 14:02:47,997 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'napalm_configure' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/.local/lib/python3.10/site-packages/nornir_napalm/plugins/tasks/napalm_configure.py", line 43, in napalm_configure
    device.commit_config()
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 573, in commit_config
    raise ReplaceConfigException(msg)
napalm.base.exceptions.ReplaceConfigException: napalm-ios replace() requires Cisco 'archive' feature to be enabled

2022-12-27 14:02:47,998 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'nornir_napalm_send_config_exmaple' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/devnet/pyhton_nornir/2023/5.nornir_napalm/5.6.nornir_napalm_send_config_replace.py", line 9, in nornir_napalm_send_config_exmaple
    task.run(task=napalm_configure, filename=f"{task.host}", replace=True)
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 174, in run
    raise NornirSubTaskError(task=run_task, result=r)
nornir.core.exceptions.NornirSubTaskError: Subtask: napalm_configure (failed)


2022-12-27 14:03:05,427 -  nornir.core -     INFO -        run() - Running task 'nornir_utils_write_file_example' with args {} on 1 hosts
2022-12-27 14:03:37,307 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 14:03:40,966 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'napalm_configure' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/.local/lib/python3.10/site-packages/nornir_napalm/plugins/tasks/napalm_configure.py", line 43, in napalm_configure
    device.commit_config()
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 573, in commit_config
    raise ReplaceConfigException(msg)
napalm.base.exceptions.ReplaceConfigException: napalm-ios replace() requires Cisco 'archive' feature to be enabled

2022-12-27 14:03:40,967 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'nornir_napalm_send_config_exmaple' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/devnet/pyhton_nornir/2023/5.nornir_napalm/5.6.nornir_napalm_send_config_replace.py", line 9, in nornir_napalm_send_config_exmaple
    task.run(task=napalm_configure, filename=f"{task.host}", replace=True)
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 174, in run
    raise NornirSubTaskError(task=run_task, result=r)
nornir.core.exceptions.NornirSubTaskError: Subtask: napalm_configure (failed)


2022-12-27 14:05:20,120 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 14:05:42,284 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 14:05:45,764 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'napalm_configure' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/.local/lib/python3.10/site-packages/nornir_napalm/plugins/tasks/napalm_configure.py", line 43, in napalm_configure
    device.commit_config()
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 573, in commit_config
    raise ReplaceConfigException(msg)
napalm.base.exceptions.ReplaceConfigException: napalm-ios replace() requires Cisco 'archive' feature to be enabled

2022-12-27 14:05:45,765 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'nornir_napalm_send_config_exmaple' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/devnet/pyhton_nornir/2023/5.nornir_napalm/5.6.nornir_napalm_send_config_replace.py", line 9, in nornir_napalm_send_config_exmaple
    task.run(task=napalm_configure, filename=f"{task.host}", replace=True)
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 174, in run
    raise NornirSubTaskError(task=run_task, result=r)
nornir.core.exceptions.NornirSubTaskError: Subtask: napalm_configure (failed)


2022-12-27 14:06:17,293 -  nornir.core -     INFO -        run() - Running task 'nornir_utils_write_file_example' with args {} on 1 hosts
2022-12-27 14:06:37,187 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 14:07:02,665 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 14:07:35,574 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 14:08:52,176 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 14:09:06,221 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'napalm_configure' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/.local/lib/python3.10/site-packages/nornir_napalm/plugins/tasks/napalm_configure.py", line 43, in napalm_configure
    device.commit_config()
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 559, in commit_config
    output = self._commit_handler(cmd)
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 467, in wrapper
    return f(self, *args, **kwargs)
  File "/home/majid/.local/lib/python3.10/site-packages/napalm/ios/ios.py", line 484, in _commit_handler
    output = self.device.send_command(cmd, expect_string=patterns)
  File "/home/majid/.local/lib/python3.10/site-packages/netmiko/utilities.py", line 596, in wrapper_decorator
    return func(self, *args, **kwargs)
  File "/home/majid/.local/lib/python3.10/site-packages/netmiko/base_connection.py", line 1691, in send_command
    raise ReadTimeout(msg)
netmiko.exceptions.ReadTimeout: 
Pattern not detected: '(?:[>##]\\s*$|.*all username.*confirm)' in output.

Things you might try to fix this:
1. Explicitly set your pattern using the expect_string argument.
2. Increase the read_timeout to a larger value.

You can also look at the Netmiko session_log or debug log for more information.



2022-12-27 14:09:06,222 - nornir.core.task -    ERROR -      start() - Host 'R1': task 'nornir_napalm_send_config_exmaple' failed with traceback:
Traceback (most recent call last):
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 99, in start
    r = self.task(self, **self.params)
  File "/home/majid/devnet/pyhton_nornir/2023/5.nornir_napalm/5.6.nornir_napalm_send_config_replace.py", line 9, in nornir_napalm_send_config_exmaple
    task.run(task=napalm_configure, filename=f"{task.host}", replace=True)
  File "/home/majid/.local/lib/python3.10/site-packages/nornir/core/task.py", line 174, in run
    raise NornirSubTaskError(task=run_task, result=r)
nornir.core.exceptions.NornirSubTaskError: Subtask: napalm_configure (failed)


2022-12-27 14:09:53,232 -  nornir.core -     INFO -        run() - Running task 'nornir_utils_write_file_example' with args {} on 1 hosts
2022-12-27 14:10:20,188 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 15:10:21,201 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 15:17:44,234 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 16:05:27,561 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 16:06:12,777 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 16:06:48,274 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 16:08:08,642 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-27 16:19:39,775 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-28 03:06:08,967 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-28 03:06:57,716 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-29 02:26:21,080 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-29 02:27:49,654 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_send_config_exmaple' with args {} on 1 hosts
2022-12-30 02:16:22,273 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_utils_backup_example' with args {} on 1 hosts
2022-12-30 02:17:21,584 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_utils_backup_example' with args {} on 1 hosts
2022-12-30 02:17:36,705 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_utils_backup_example' with args {} on 1 hosts
2022-12-30 02:21:32,045 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_restore_with_send_config_replace' with args {} on 1 hosts
2022-12-30 02:26:35,256 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_utils_backup_example' with args {} on 1 hosts
2022-12-30 02:29:01,739 -  nornir.core -     INFO -        run() - Running task 'nornir_napalm_restore_with_send_config_replace' with args {} on 1 hosts
